tensorrt_llm/_torch/attention_backend/vanilla.py

```
@@ -399,10 +467,33 @@ def forward(self,
-        ]
-        kv_cache_tensor = metadata.kv_cache_manager.get_buffers(self.layer_idx)
+        if self._use_turboquant4_kv_cache():
+            assert metadata.request_ids is not None
+            key_cache_indices = metadata.kv_cache_manager.get_batch_cache_indices_for_role(
+                metadata.request_ids, self.layer_idx, Role.KEY)
+            value_cache_indices = metadata.kv_cache_manager.get_batch_cache_indices_for_role(
+                metadata.request_ids, self.layer_idx, Role.VALUE)
+            scale_cache_indices = metadata.kv_cache_manager.get_batch_cache_indices_for_role(
+                metadata.request_ids, self.layer_idx, Role.VALUE_BLOCK_SCALE)
+            cache_indices = list(
+                zip(key_cache_indices, value_cache_indices, scale_cache_indices))
```

Ruff (0.15.13)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)